main.go

```diff
@@ -445,6 +480,122 @@ func deleteRun(_ *cobra.Command, args []string) {
 	fmt.Fprintf(stdout, "Successfully deleted secret if it existed: %s\n", object)
 }
 
+func isEditorAvailable(e string) (bool, error) {
+	cmd := exec.Command("which", "-s", e)
+	if err := cmd.Start(); err != nil {
+		return false, err
+	}
+	if err := cmd.Wait(); err != nil {
+		if _, ok := err.(*exec.ExitError); ok {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+func findEditor() error {
+	if editor == "" {
+		editor = os.Getenv("VISUAL")
+	}
+	if editor == "" {
+		editor = os.Getenv("EDITOR")
+	}
+	if editor != "" {
+		return nil
+	}
+
+	editorsToTry := []string{"vi", "emacs", "nano", "pico"}
+	for _, e := range editorsToTry {
+		if available, err := isEditorAvailable(e); err != nil {
+			return err
+		} else if available {
+			editor = e
+			return nil
+		}
+	}
+	return errors.New("unable to determine editor. manually specify using `--editor` flag and try again")
+}
+
+func editRun(_ *cobra.Command, args []string) {
+	bucket, object, err := parseRef(args[0])
+	if err != nil {
+		handleError(err, 2)
+	}
+
+	ctx := cliCtx()
+	client, err := berglas.New(ctx)
+	if err != nil {
+		handleError(err, 2)
+	}
+	var secret *berglas.Secret
+	if secret, err = client.Access(ctx, &berglas.AccessRequest{
+		Bucket: bucket,
+		Object: object,
+	}); err != nil {
+		handleError(err, 1)
+	}
+
+	if err = findEditor(); err != nil {
+		handleError(err, 1)
+	}
+
+	f, err := ioutil.TempFile("", "berglas-")
+	if err != nil {
+		handleError(errors.Wrap(err, "failed to create tempfile for secret"), 1)
+	}
+	if _, err := f.Write(secret.Plaintext); err != nil {
+		handleError(errors.Wrap(err, "failed to write tempfile for secret"), 1)
+	}
+	if err := f.Sync(); err != nil {
+		handleError(errors.Wrap(err, "failed to sync tempfile for secret"), 1)
+	}
+	if err := f.Close(); err != nil {
+		handleError(errors.Wrap(err, "failed to close tempfile for secret"), 1)
+	}
+
+	cmd := exec.Command(editor, f.Name())
+	cmd.Stdout = stdout
+	cmd.Stderr = stderr
+	if err := cmd.Start(); err != nil {
+		handleError(errors.Wrap(err, "failed to start editor"), 2)
+	}
+	if err := cmd.Wait(); err != nil {
+		if _, ok := err.(*exec.ExitError); ok {
+			handleError(errors.Wrap(err, "editor failed to exit cleanly"), 2)
+		}
+		handleError(errors.Wrap(err, "unknown failure in running editor"), 2)
+	}
+
+	newPlaintext, err := ioutil.ReadFile(f.Name())
+	if err != nil {
+		// try to cleanup
+		_ = os.Remove(f.Name())
+		handleError(errors.Wrapf(err, "failed to read secret tempfile"), 2)
+	}
+
+	if err = os.Remove(f.Name()); err != nil {
+		handleError(errors.Wrapf(err, "failed to delete secret tempfile"), 2)
+	}
+
+	if bytes.Equal(secret.Plaintext, newPlaintext) {
```

Comment:
We should return an error here saying the data was unchanged. Or at the very least, exit 0, but print a warning.